src/tracks/playabletrack/wavetrack/ui/SpectrumCache.cpp

```diff
@@ -369,8 +373,10 @@ void SpecCache::Populate(
    const size_t fftLen = windowSizeSetting * zeroPaddingFactorSetting;
    const auto nBins = settings.NBins();
 
+   // One work area for pffft, one work area for the outputs, two extra
+   // output areas to compute reassignment
    const size_t scratchSize =
-      reassignment ? 3u * PffftAlignedCount{ fftLen } : fftLen;
+      (reassignment ? 4u : 2u) * PffftAlignedCount{ fftLen };
```

Comment:
We're here in `Populate`, and within that function it's difficult to find out why there's a need for more than just `fftLen`. One has to dig `CalculateOneSpectrum` to find out about that dependency. And in fact, it doesn't seem that the standard spectrogram view requires `scratch`.

Reply:
I think the comment before this line is sufficient explanation.